src/audio/code_8019AF00.c

```diff
@@ -4211,46 +4265,246 @@ void AudioSfx_LowerSfxSettingsReverb(Vec3f* pos, s8 isReverbLowered) {
     }
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019F5AC.s")
+f32 Audio_SetSyncedSfxFreqAndVolume(f32 freqVolParam) {
+    f32 ret = 1.0f;
+
+    if (freqVolParam > 6.0f) {
+        sSfxSyncedVolume = 1.0f;
+        sSfxSyncedFreq = 1.1f;
+    } else {
+        ret = freqVolParam / 6.0f;
+        sSfxSyncedVolume = (ret * 0.22500002f) + 0.775f;
+        sSfxSyncedFreq = (ret * 0.2f) + 0.9f;
+    }
+
+    return ret;
+}
+
+/**
+ * Adjusts both frequency and volume based on a single parameter "freqVolParam"
+ * When freqVolParam >= 6.0f, frequency is increased to 1.1f and volume remains fixed at 1.0f
+ * For every -1.0f taken from freqVolParam (eg. 5.0f, 4.0f, 3.0f...):
+ *     - volume will decrease by 0.0375f
+ *     - frequency will decrease by 0.0333333f
+ */
+void Audio_PlaySfx_AtPosForMetalEffectsWithSyncedFreqAndVolume(Vec3f* pos, u16 sfxId, f32 freqVolParam) {
+    f32 sp2C;
+    f32 phi_f0;
+    s32 phi_v0;
+    u16 metalSfxId = 0;
+
+    sp2C = Audio_SetSyncedSfxFreqAndVolume(freqVolParam);
+    AudioSfx_PlaySfx(sfxId, pos, 4, &sSfxSyncedFreq, &sSfxSyncedVolume, &gSfxDefaultReverb);
```

Comment:
```suggestion
    f32 sp2C = Audio_SetSyncedSfxFreqAndVolume(freqVolParam);
    f32 phi_f0;
    s32 phi_v0;
    u16 metalSfxId = 0;

    AudioSfx_PlaySfx(sfxId, pos, 4, &sSfxSyncedFreq, &sSfxSyncedVolume, &gSfxDefaultReverb);
```

Reply:
Since you're changing setting a 0 value from before a function call to after a function call, it doesn't match. I did use `NA_SE_NONE` for the sfxid though

Reply:
Oh you are right, I missed that :sweat: